src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -1196,7 +1202,12 @@ impl<'a, 'b> BuildReducedGraphVisitor<'a, 'b> {
                 self.r.define(module, ident, MacroNS, (res, vis, span, expansion, IsMacroExport));
             } else {
                 self.r.check_reserved_macro_name(ident, res);
-                self.insert_unused_macro(ident, item.id, span);
+                self.insert_unused_macro(
+                    ident,
+                    self.r.definitions.local_def_id(item.id),
```

Comment:
`let def_id = self.r.definitions.local_def_id(item.id);` is already evaluated above.
You can move it slightly higher and reuse in all the places inside `fn insert_unused_macro`.

Reply:
Fixed in https://github.com/rust-lang/rust/pull/72750